### PR TITLE
Handle readme and license files in scarb package

### LINKED ISSUE #737: Handle Readme and license files in `scarb package`
### Problem

Add README and LICENSE files based on what's in Package manifest here:

https://github.com/software-mansion/scarb/blob/b78deefe94353f35a86b5ff991db4a44ce1d1ddf/scarb/src/core/publishing/source.rs#L23-L26

Add tests by fixing this TODO: https://github.com/software-mansion/scarb/blob/0951e2a12bbfc41a91ad359c48d2e0be51217b6c/scarb/tests/package.rs#L545

### Proposed Solution

_No response_

### Notes

Do not forget that README and LICENSE files may be outside package root.

### PR DESCRIPTION
**Pull request by szymmis**
Resolves #737

### CODE REVIEW

File: scarb/src/core/manifest/toml_manifest.rs [hunk 1]
```diff
@@ -814,6 +838,18 @@ impl TomlManifest {
...
+pub fn license_for_package(
+    package_root: &Utf8Path,
+    license_file: Option<&Utf8PathBuf>,
+) -> Result<Option<Utf8PathBuf>> {
+    if let Some(p) = license_file {
+        Ok(abs_canonical_path(package_root, Some(p.as_path()))?)
+    } else {
+        Ok(None)
+    }
+}
```
Comment (maciektr) [inline scarb/src/core/manifest/toml_manifest.rs:851]: Isn't this equivalent to:
```
abs_canonical_path(package_root, license_file.map(|p| p.as_path())
```
?

Maybe wrapping second argument of abs_canonical_path with AsRef would help?

File: scarb/src/core/manifest/toml_manifest.rs [hunk 2]
```diff
@@ -832,14 +868,14 @@ pub fn readme_for_package(
...
 
-/// Creates the absolute canonical path of the README file and checks if it exists
-fn abs_canonical_path(prefix: &Utf8Path, readme: Option<&Utf8Path>) -> Result<Option<Utf8PathBuf>> {
-    match readme {
+/// Creates the absolute canonical path of the file and checks if it exists
+fn abs_canonical_path(prefix: &Utf8Path, path: Option<&Utf8Path>) -> Result<Option<Utf8PathBuf>> {
+    match path {
         None => Ok(None),
         Some(readme) => {
             let path = prefix.parent().unwrap().join(readme);
```
Comment (maciektr) [inline scarb/src/core/manifest/toml_manifest.rs:876]:
```suggestion
        Some(path) => {
            let path = prefix.parent().unwrap().join(path);
```

File: scarb/src/core/manifest/toml_manifest.rs [hunk 3]
```diff
@@ -140,10 +140,26 @@ impl PackageInheritableFields {
...
+        workspace_root: &Utf8Path,
+        package_root: &Utf8Path,
+    ) -> Result<Utf8PathBuf> {
+        let Some(license_file) = license_for_package(workspace_root, self.license_file.as_ref())?
+        else {
+            bail!("no `license_file` found in workspace definition");
+        };
+
+        diff_utf8_paths(
+            workspace_root.parent().unwrap().join(license_file),
```
Comment (maciektr) [inline scarb/src/core/manifest/toml_manifest.rs:157]: Isn't `license_file` already absolute and canonical? 🤔

File: scarb/src/core/publishing/source.rs [hunk 4]
```diff
@@ -53,6 +58,11 @@ fn push_worktree_files(pkg: &Package, ret: &mut Vec<Utf8PathBuf>) -> Result<()>
                 return false;
             }
 
+            // Skip README and LICENSE files
+            if path == readme || path == license_file {
```
Comment (maciektr) [inline scarb/src/core/publishing/source.rs:62]: Are `readme` and `license_file` absolute here?

Comment (szymmis) [reply scarb/src/core/publishing/source.rs:62]: Yes, they are

File: scarb/tests/package.rs [hunk 5]
```diff
@@ -754,7 +754,126 @@ fn list_ignore_nested() {
         "#}));
 }
 
-// TODO(mkaput): Invalid readme/license path
+#[test]
+fn include_readme_and_license() {
```
[the remark this answers is not in the document]
Comment (mkaput) [reply scarb/tests/package.rs:758]: or please add the README file here :)

**Review by mkaput — APPROVED** [review]
👏🏻

**Pull request merged**